### PR TITLE
database: store module decoders inside database

### PR DESCRIPTION
**Pull request by Maan2003**
Store ModuleDecoderRegistry inside Database, so it doesn't need to be supplied begin_transaction.

Split off from #1197

### CODE REVIEW

File: fedimint-api/src/db/mem_impl.rs [hunk 1]
```diff
@@ -157,59 +155,64 @@ impl Iterator for MemDbIter {
 #[cfg(test)]
 mod tests {
     use super::MemDatabase;
+    use crate::{db::Database, module::registry::ModuleDecoderRegistry};
+
+    fn database() -> Database {
```
Comment (m1sterc001guy) [inline fedimint-api/src/db/mem_impl.rs:160]: nit: change to new_database()?

**Review by m1sterc001guy — APPROVED** [review]
LGTM

**Review by dpc — APPROVED** [review]

File: integrationtests/tests/fixtures/mod.rs [hunk 2]
```diff
@@ -284,7 +285,8 @@ pub async fn fixtures(num_peers: u16) -> anyhow::Result<Fixtures> {
                                 "wallet",
                                 Wallet::new_with_bitcoind(
                                     cfg.get_module_config_typed("wallet").unwrap(),
-                                    fed_db(),
+                                    // wallet module does not need any module specific types
```
Comment (dpc) [inline integrationtests/tests/fixtures/mod.rs:288]: Ah, I've overlooked it. 

I'm not so sure about it...

It has inputs, outputs, etc.

Comment (dpc) [reply integrationtests/tests/fixtures/mod.rs:288]: We should pass the `&Database` to the closure here. I'll have this in #1269

Comment (Maan2003) [reply integrationtests/tests/fixtures/mod.rs:288]: I think this should work. 
Current this db is only passed to broadcast pending task, which also used default decoders before.

**Pull request merged**